core/src/test/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationServiceTest.java

```diff
@@ -330,12 +345,21 @@ public void testUpdateNodeCountStartInstances(String methodName, DetailedStackSt
         stack.setId(STACK_ID);
         stack.setStackStatus(new StackStatus(stack, stackStatus));
         StackDto stackDto = mock(StackDto.class);
+        setupMocksForStopStartInstanceGroupValidation(stackDto);
+        CmTemplateProcessor cmTemplateProcessor = mock(CmTemplateProcessor.class);
+
         lenient().when(stackDto.getId()).thenReturn(STACK_ID);
         when(stackDto.getStack()).thenReturn(stack);
+        when(cmTemplateProcessorFactory.get(anyString())).thenReturn(cmTemplateProcessor);
 
         InstanceGroupAdjustmentV4Request upscaleAdjustment = new InstanceGroupAdjustmentV4Request();
         upscaleAdjustment.setScalingAdjustment(5);
+        upscaleAdjustment.setInstanceGroup("compute");
+
+        ReflectionTestUtils.setField(updateNodeCountValidator, "cmTemplateProcessorFactory", cmTemplateProcessorFactory);
```

Comment:
First question: why the `updateNodeCountValidator` is a `Spy` instead of being `Mock`

Reply:
Not sure why it was a `Spy`. Refactored to use `Mock`.